core/src/test/scala/unit/kafka/server/GroupCoordinatorBaseRequestTest.scala

```diff
@@ -0,0 +1,158 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.server
+
+import kafka.test.ClusterInstance
+import kafka.utils.{NotNothing, TestUtils}
+import org.apache.kafka.common.message.{ConsumerGroupHeartbeatRequestData, JoinGroupRequestData, OffsetCommitRequestData, OffsetCommitResponseData, SyncGroupRequestData}
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.requests.{AbstractRequest, AbstractResponse, ConsumerGroupHeartbeatRequest, ConsumerGroupHeartbeatResponse, JoinGroupRequest, JoinGroupResponse, OffsetCommitRequest, OffsetCommitResponse, SyncGroupRequest, SyncGroupResponse}
+import org.junit.jupiter.api.Assertions.assertEquals
+
+import scala.jdk.CollectionConverters._
+import scala.reflect.ClassTag
+
+class GroupCoordinatorBaseRequestTest(cluster: ClusterInstance) {
```

Comment:
Is the goal here to create a new baseRequestTest class? I noticed we didn't extend the existing BaseRequestTest class

Reply:
We can't reuse `BaseRequestTest` when using `@ClusterTest`. Here, I created `GroupCoordinatorBaseRequestTest` to group the common functionalities required to test Group Coordinator Apis.

Reply:
I see. The naming reminds me of BaseRequest test and we have the same/similar `connectAndReceive` method. I think this is fine, but these test files do start to get confusing when we have so many similarly named but very different classes. Not something we need to address in this PR though.

Reply:
Yeah, I agree. I was considering using a test context class instead of using a base class. I am not sure if it would really be better though... What do you think?

Reply:
I think it is ok as it is. I wonder if naming helps? Either way we can do as a followup